clients/drcachesim/tests/analysis_unit_tests.cpp

```diff
@@ -177,10 +183,184 @@ test_queries()
     return true;
 }
 
+bool
+test_wait_records()
+{
+#ifdef HAS_ZIP
+    std::cerr << "\n----------------\nTesting wait records\n";
+
+    static constexpr int NUM_INPUTS = 5;
+    static constexpr int NUM_OUTPUTS = 2;
+    static constexpr int NUM_INSTRS = 9;
+    static constexpr memref_tid_t TID_BASE = 100;
+    static constexpr int CPU0 = 6;
+    static constexpr int CPU1 = 9;
+    std::vector<trace_entry_t> inputs[NUM_INPUTS];
+    for (int i = 0; i < NUM_INPUTS; i++) {
+        memref_tid_t tid = TID_BASE + i;
+        inputs[i].push_back(make_thread(tid));
+        inputs[i].push_back(make_pid(1));
+        // The last input will be earlier than all others. It will execute
+        // 3 instrs on each core. This is to test the case when an output
+        // begins in the wait state.
+        for (int j = 0; j < (i == NUM_INPUTS - 1 ? 6 : NUM_INSTRS); j++)
+            inputs[i].push_back(make_instr(42 + j * 4));
+        inputs[i].push_back(make_exit(tid));
+    }
+
+    // Synthesize a cpu-schedule file with some waits in it, if run in lockstep.
```

Comment:
Add what the dots and dashes mean for a quick reference for the reader. Dashes are waits and dots are non-instrs?

Reply:
Yes, moved the constant strings into the comment as what would happen with pure lockstep.